pvl-tools/pvl.cson

```diff
@@ -30,67 +30,104 @@ services:
-      , attr: "href"
-      , into: "username_link"
-      , error: ["FAILED_PARSE", "Couldn't find username href"] } },
-    { parse_url: {
-      , from: "username_link"
-      , path: "path"
-      , error: ["FAILED_PARSE", "Failed to parse username URL: %{username_link}"] } },
-    { regex_capture: {
-      , pattern: "^[^/]*/([^/]*)$"
-      , from: "path"
-      , into: ["split_path_1"]
-      , error: ["FAILED_PARSE", "Username URL has no path"] } },
-    { assert_compare: {
-      , cmp: "stripdots-then-cicmp"
-      , a: "split_path_1"
-      , b: "username_service"
-      , error: ["BAD_USERNAME", "Usernames don't match '%{split_path_1}' vs '%{username_service}'"] } },
-    # check the proof text
+      , selectors: ["code", 0, {contents: true}, 0]
+      , into: "first_code_comment"
+      , data: true
+      , error: ["FAILED_PARSE", "Could not find proof markup comment in Facebook's response"] } },
+
+    # Facebook escapes "--" as "-\-\" and "\" as "\\" when inserting text into
+    # comments. Unescape these.
+    { replace_all: {
+      , old: "-\\-\\"
+      , new: "--"
+      , from: "first_code_comment"
+      , into: "fcc2" } },
+    { replace_all: {
+      , old: "\\\\"
+      , new: "\\"
+      , from: "fcc2"
+      , into: "fcc3" } },
+
+    # Load the de-escaped comment as html
+    { parse_html: {
+      , from: "fcc3"
+      , error: ["FAILED_PARSE", "Failed to parse proof markup comment in Facebook post: %{fcc3}"] } },
+
+    # This query operates on the newly extractaparsed comment, not the original page load
+    # This is the selector for the post attachment links, which contains the
+	  # proof text. It's the "<a> tags inside the div that's the immediate
+	  # *sibling* of the 'userContet' div". The second of these three <a> tags
+	  # contains the proof text, the others are blank. But we just check their concatenation.
     { selector_css: {
-      , selectors: ["#m_story_permalink_view > div:first-child > div:first-child > div:first-child h3", 1]
-      , into: "header"
-      , error: ["FAILED_PARSE", "Couldn't find proof text header"] } },
+      , selectors: ["div.userContent+div a"]
+      , multi: true
+      , into: "link_text"
+      , error: ["FAILED_PARSE", "Could not find link text in Facebook's response"] } },
     { whitespace_normalize: {
-      , from: "header"
-      , into: "header_nw" } },
+      , from: "link_text"
+      , into: "link_text_nw" } },
+
+    # Check the link text for username and sig
     { regex_capture: {
-      , pattern: "^Verifying myself: I am (\\S+) on Keybase\\.io\\. %{sig_id_medium}$"
-      , from: "header_nw"
-      , into: ["username_from_header"]
-      , error: ["TEXT_NOT_FOUND", "Proof text not found: 'Verifying myself: I am %{username_keybase} on Keybase.io. %{sig_id_medium}' != '%{header_nw}'"] } },
+      , pattern: "^.*Verifying myself: I am (\\S+) on Keybase.io. (\\S+).*$"
```

Comment:
I think I'm stricter than this in Go/Iced. I don't allow any characters before and after the proof text, except maybe whitespace. Should we keep the same approach here?